src/storage/filesystem-adapter.test.ts

```
@@ -0,0 +1,150 @@
+  });
+
+  describe('constructor', () => {
+    it('should use default file path if not provided', () => {
+      const defaultAdapter = new FilesystemAdapter();
+      expect(defaultAdapter['filePath']).toBe(path.join(process.cwd(), 'run-cache-data.json'));
+    });
+
+    it('should use custom file path if provided', () => {
+      const customPath = path.join(process.cwd(), 'custom-cache.json');
+      const customAdapter = new FilesystemAdapter({ filePath: customPath });
+      expect(customAdapter['filePath']).toBe(customPath);
```

ESLint

[error] ["filePath"] is better written in dot notation.

(dot-notation)